src/lib.rs

```diff
@@ -34,6 +34,13 @@ use std::convert::{TryFrom, TryInto};
 // We are able to convert the proto on read, so the tools can still read version 9/10/11.
 pub const CURRENT_VERSION: i32 = 12;
 
+/// size for batch
+pub const NUM_POINTS_PER_BATCH: usize = 500_000;
+
+pub trait NumberOfPoints {
```

Comment:
This is going to be used in `generation.rs`, where the number of points is queried, but once we have iterators over batches, the `size_hint()` of an iterator can't be used any more to query the number of points.

Reply:
Who else will implement this trait?

Reply:
`query_frustum.rs`: `impl NumberOfPoints for Points`
`generation.rs`: `impl NumberOfPoints for Points` and `where P: Iterator<Item = PointsBatch> + NumberOfPoints,`
`octree_test.rs`: `impl<T> NumberOfPoints for std::vec::IntoIter<T>`
`input_file_iterator.rs`: `impl NumberOfPoints for InputFileIterator`
`node_iterator.rs`: `impl NumberOfPoints for NodeIterator`
`ply.rs` and `pts.rs`: `impl NumberOfPoints for {Ply|Pts}Iterator`

Reply:
Okay. That could maybe be the start for a `PointsReader` trait or similar, right?

Reply:
It could be a `PointsBatchIterator` trait, but since not all iterators need to know the number of points and since it may be useful somewhere else, too, I just kept it separate...